aws/resource_aws_s3_bucket_analytics_configuration.go

```diff
@@ -147,19 +148,23 @@ func resourceAwsS3BucketAnalyticsConfigurationPut(d *schema.ResourceData, meta i
 
 	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
 		_, err := s3conn.PutBucketAnalyticsConfiguration(input)
+
+		if tfawserr.ErrCodeEquals(err, s3.ErrCodeNoSuchBucket) {
```

Comment:
open q: why move this if-logic outside of the `if err != nil` check below? just curious b/c i've caught myself writing these types of checks w/in an `if err != nil` especially in some Delete CRUD ops of resources and i think my rationale at the time was that it didn't seem to make sense to pass the err around if it's `nil` but maybe that goes against our usual patterns...

Reply:
Great question! It's something I picked up from https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

In this case since it is the unhappy path being nested under the `if err != nil`, it is probably _okay_ to be nested if we wanted. Prior to these AWS SDK Go error checking helpers it used to be that this logic _had_ to be nested (or duplicate the `err != nil` check) for the type assertions, but now that this detail is tucked away, it this allows for either style. 👍

Reply:
Ohh very cool, thanks! def sounds familiar coming from ruby land (and Sandi Metz's principles)